pkg/v1/tkg/test/tkgctl/shared/e2e_common.go

```diff
@@ -223,3 +241,84 @@ func E2ECommonSpec(ctx context.Context, inputGetter func() E2ECommonSpecInput) {
 		By("Test successful !")
 	})
 }
+
+func GetNextAvailableTkrVersion(tkgctlClient tkgctl.TKGClient, currentTkrVersion string) (string, error) {
+	tkrs, err := tkgctlClient.GetTanzuKubernetesReleases("")
+	if err != nil {
+		return "", err
+	}
+
+	foundTkrVersion := ""
+	availableVersions := make(map[string]string)
+	for i := range tkrs {
+		compatible := ""
+		for _, condition := range tkrs[i].Status.Conditions {
+			if condition.Type == runv1alpha3.ConditionCompatible {
+				compatible = string(condition.Status)
+				break
+			}
+		}
+		if !strings.EqualFold(compatible, "true") {
+			continue
+		}
+		labels := tkrs[i].Labels
+		if labels != nil {
+			if _, exists := labels[runv1alpha3.LabelDeactivated]; exists {
+				continue
+			}
+		}
+
+		compareResult, err := utils.CompareVMwareVersionStrings(currentTkrVersion, tkrs[i].Spec.Version)
+		if err != nil {
+			return "", err
+		}
+
+		if compareResult < 0 {
+			if foundTkrVersion == "" {
+				foundTkrVersion = tkrs[i].Spec.Version
+			} else {
+				compareResult, err := utils.CompareVMwareVersionStrings(foundTkrVersion, tkrs[i].Spec.Version)
+				if err != nil {
+					return "", err
+				}
+				if compareResult > 0 {
+					foundTkrVersion = tkrs[i].Spec.Version
+				}
+			}
+			availableVersions[tkrs[i].Spec.Version] = tkrs[i].Name
```

Comment:
what is `availableVersions` use for?

Reply:
not used. Left over from last refactoring. removing.